Developer

makires opened the pull request:
Test pull request through GitHub. 
MainCopy is empty start project, developer is current work branch.

Commit:
main view - start test commit

Commit:
Set paddings and frames for City Title and Current Weather Views in Header Weather View

Commit:
Alignment and spacing elements in Header View

Commit:
Row for List Raily weather through VStack and HStack

Commit:
frame 44x44 for temp2 in RowListDailyWeather resolve problem with alignment

Commit:
refactoring rows for ListDailyWeatherView

lukvan committed:
Rename image for CurrentWeatherView = Cloudy

lukvan committed:
move backgroundImage to ZStack for HeaderWeatherView and change padding for ZStack

lukvan committed:
Change frame and padding for CurrentWeatherView = Temp + Description.

lukvan committed:
border for image in CurrentWeaherView

lukvan committed:
Refactoring Details For CurrentWeatherView in ScrollView and change padding

lukvan committed:
Ready to show first Screen, with comments borders and Note comments

lukvan committed:
Alignment code

lukvan committed:
Adding edgesIgnoringSafeArea for Image in HeaderView, which resolve problem with border on Safe Area

lukvan committed:
Creating .gitignore

lukvan committed:
Commit before branching SecondView

lukvan committed:
Use GeometryReader to help layout views in a relative manner

lukvan committed:
Change frames and sizes for CityTitleView throughout GeometryReader

lukvan committed:
GeometryReader broke MainView into ZStack, width CityTitleVIew change to infinity

lukvan committed:
stuck

lukvan committed:
Cleaning

Commit:
Refactoring structure Main

Commit:
Start changing DetailsForCurrentWeatherView

lukvan committed:
Refactoring

lukvan committed:
List with .offset(y: -110)

Commit:
attempt to change offset List through PreferenceKey Protocol

Commit:
install Alamofire

Commit:
start learning and testing Alamofire

lukvan committed:
api service from weatherapi.com

lukvan committed:
rename repeated element in states collections

Commit:
creating Network Layer through Alamofire

lukvan committed:
Change List on ScrollView for test with offset

Commit:
Transfer coordinate Y left top point to through @State and @Binding

lukvan committed:
удаление Binding для проверки работоспособности

lukvan committed:
Animation for background Color.clear-Color.white in Details Current Weather

Commit:
add model Weather foe ViewModel. Change CurrentWeatherView appropriate properties. Add items for APIWeather

Commit:
creating new branch for test help tools. paw and quickType
удобная вещь по созданию быстрой модели API для парсинга данных. PAW поддерживает запросы для Alamofire

Commit:
generated file APIHourly through quick Type with terminal

lukvan committed:
second model for hourly weather

Commit:
parsing daily, hourly weather through two separate(but the same) models.
bug: parsing only three daily weather.
bug: bad scrolling
bug: image on header broken because Geometry Reader

Commit:
before change APImodel. Delete json dummy

Commit:
clean project before localisation

Commit:
test localizbale, bug with string interpolation

lukvan committed:
creating folder Subview for main display.
tip: install SwiftLint , make simple Localizable.

lukvan committed:
pod install SwiftLint

lukvan committed:
Add SwiftLint through Pods, delete warnings from SwiftLint.

Commit:
некоторые файлы от Pods не комитятся для проекта: .swiftlint.yml. try to commit all files

Commit:
некоторые файлы от Pods не комитятся для проекта: .swiftlint.yml. try to commit all files

Commit:
не удается экспортировать локализация. ошибка из-за Alamofire during build. Попробую без pod files

Commit:
single function request for weatherVM. Add new function into weatherVM.
tip: think about async-await for without closure-hell.
constant variable for localisation in CurrentWeatherView through HStack.
Add Localizable for En and Ru

lukvan committed:
add variable observing localization phone for choose request

lukvan committed:
change localization for wind.
get current Locale and generate request on server.

lukvan committed:
modifier .scaledToFill into Header..View is a bug. Without all work good.
LeftTopPont from ScrollView send coordinate minY to ViewModel, and we can observe this property into Header, where we're needs get changes for frame, opacity and etc.
Delete all @Bindings, it was bad idea. Summary: the view can send value to viewModel from .onPreferenceChange(), and other views can see this changes through  viewModel. I think it is good.

lukvan committed:
Localization for dynamic current city Title. Note, request static city, but view dynamic (from viewModel).
Left top point from Daily Weather now static CGFloat, think about flexible.
Change ApiWeather model, add location for current weather.

lukvan committed:
Hourly current weather is done: separate item for current hour and other hours through ForEach with dropFirst.
Add localization for current hour like Now(Сейчас)

Commit:
Add localisation for  all wind direction and pressure.
Change model for icon (rename iconURL and add code icon for system name Image through  switch - cases. QA: можно ли это как-то сделать лучше и быстрее? Because very tedious.
Add image SF Symbol for hourly and daily weather.
API Model add currentLocalTime appropriate local. Good for other regions and current data. (solved bug with midnight and timezone.

lukvan committed:
Change converting for temperature: add º, + through computed property.
Add animation, but deprecated modifier. Think about and learn.
Bug: for dateFormatter not work doesRelativeDateFormatting with .dateFormatter and dateStyle.

Commit:
Add VStack for details Current Weather, because long weatherText broke alignment.
Add logic to dateFormatter for dailyWeather (today, tomorrow , сегодня, завтра).
Question: server and local time on device conflict to dailyWeather.

lukvan committed:
Add alignment top for ZStack in HourlyWeather.
Bug: bot alignment temperature, when receive big icon.

lukvan committed:
Clear with SwiftLint.

lukvan committed:
merge with main

ismail9001 commented on Podfile:
```diff
@@ -0,0 +1,11 @@
+  use_frameworks!
+
+  # Pods for WeatherProjectTestOrion
+  pod 'Alamofire', '~> 5.5'
```
добавить поды в гит игнор

makires replied:
Done. Pods ignored. 
commit 966a197

ismail9001 commented on WeatherProjectTestOrion/Assets.xcassets/cloudyBackground.imageset/Contents.json:
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "filename" : "zrj-TPjcRLA.png",
```
используй понятные названия в названиях картинок (все проверить)

makires replied:
Done. Commit 136b829. Other images has correct names.

ismail9001 commented on WeatherProjectTestOrion/Main/Model/HourlyCurrentWeather.swift:
```diff
@@ -0,0 +1,29 @@
+    let temp: String
+}
+
+let hourlyCurrentWeather = [
```
это ведь уже не нужно?

makires replied:
No, file deleted. It was dummy model.
Commit 76b8ae5.

ismail9001 commented on WeatherProjectTestOrion/Main/Model/ListDailyWeather.swift:
```diff
@@ -0,0 +1,53 @@
+    let temp2: String
+}
+
+let listDaily = [
```
это нужно?

makires replied:
Commit 76b8ae5 also.

ismail9001 commented on WeatherProjectTestOrion/Main/Model/ListDailyWeather.swift:
```diff
@@ -0,0 +1,53 @@
+    let nameDay: String
+    let icon: String
+    let colorIcon: Color
+    let temp1: String
```
непонятные названия тампературы

makires replied:
Commit 76b8ae5.
I'll remember about naming.

ismail9001 commented on WeatherProjectTestOrion/Main/Model/ListDailyWeather.swift:
```diff
@@ -0,0 +1,53 @@
+
+struct DayWeather: Hashable {
+    let date: String
+    let nameDay: String
```
просто day

makires replied:
File deleted. 
Commit 76b8ae5 also.

ismail9001 commented on WeatherProjectTestOrion/Main/Model/Weather.swift:
```diff
@@ -0,0 +1,66 @@
+    var temperatureCurrent = ""
+    var pressureHPa  = ""
+    var windKph  = ""
+    var windDirection = ""
```
используй enum у тебя ограниченное количество направлений

makires replied:
enum для направления ветра добавила, использую его при парсинге в апи. 
По части строкового значения для ветра в этой модели, необходимы разъяснения. Тут как с температурой, дефолтное значение будет пустой строкой, если ответ от сервера не придет. 🙄
82afa136bc427741590699808e9ae84951d6aa4b

makires replied:
Done.

ismail9001 commented on WeatherProjectTestOrion/Main/Model/Weather.swift:
```diff
@@ -0,0 +1,66 @@
+
+struct Weather {
+    var cityName = ""
+    var temperatureCurrent = ""
```
число

makires replied:
Эта модель используется в текстовых элементах во вью, если поставить число, то придется делать опционал (либо инициализировать какую-нибудь дефолтную модель для response API).  В любом случае, если будет nil, надо что-то показать, а показывать число 0 в температуре, когда температуры совсем нет, не лучший вариант, или я ошибаюсь в понимании комментария? И еще эта модель инициализируется с этими свойствами, и она @Published во вью модель. Я попробовала поменять, там получается полный хаос во вью и со string interpolation 😬 
Text("\(weather.current.temperatureCurrent ?? 0 )")
Оставить как есть или покопаться и заменить?

ismail9001 commented on WeatherProjectTestOrion/Main/Model/Weather.swift:
```diff
@@ -0,0 +1,66 @@
+struct Weather {
+    var cityName = ""
+    var temperatureCurrent = ""
+    var pressureHPa  = ""
```
число

makires replied:
also https://github.com/makires/WeatherProjectTestOrion/pull/2#discussion_r776018866

ismail9001 commented on WeatherProjectTestOrion/Main/Model/Weather.swift:
```diff
@@ -0,0 +1,66 @@
+    var cityName = ""
+    var temperatureCurrent = ""
+    var pressureHPa  = ""
+    var windKph  = ""
```
число

makires replied:
also https://github.com/makires/WeatherProjectTestOrion/pull/2#discussion_r776018866

ismail9001 commented on WeatherProjectTestOrion/Main/Model/Weather.swift:
```diff
@@ -0,0 +1,66 @@
+    var windDirection = ""
+    var humidity  = ""
+    var feelsLikeTemperature  = ""
+    var text = ""
```
что за текст?

makires replied:
replace textWeatherCondition. Done.
Commit a19309a713bd25b5ed175bf0e8a9b6415461e5d8

ismail9001 commented on WeatherProjectTestOrion/Main/Model/Weather.swift:
```diff
@@ -0,0 +1,66 @@
+    var pressureHPa  = ""
+    var windKph  = ""
+    var windDirection = ""
+    var humidity  = ""
```
число

makires replied:
also https://github.com/makires/WeatherProjectTestOrion/pull/2#discussion_r776018866

ismail9001 commented on WeatherProjectTestOrion/Main/Model/Weather.swift:
```diff
@@ -0,0 +1,66 @@
+    var windKph  = ""
+    var windDirection = ""
+    var humidity  = ""
+    var feelsLikeTemperature  = ""
```
число

makires replied:
also https://github.com/makires/WeatherProjectTestOrion/pull/2#discussion_r776018866

ismail9001 commented on WeatherProjectTestOrion/Main/Model/Weather.swift:
```diff
@@ -0,0 +1,66 @@
+    var humidity  = ""
+    var feelsLikeTemperature  = ""
+    var text = ""
+    var icon = ""
```
енум

makires replied:
не понимаю до конца как сделать иконку через enum.
иконка погоды основывается на Int который приходит от сервера. Я делаю switch, так как использовать число в case внутри enum нельзя. Сведения об иконке погоды анализируются через extension ConditionWeather и уже на основе визуального сходства я выбирала символы из sf symbols.  Нужен совет

ismail9001 commented on WeatherProjectTestOrion/Main/Model/Weather.swift:
```diff
@@ -0,0 +1,66 @@
+    var feelsLikeTemperature  = ""
+    var text = ""
+    var icon = ""
+    init() { }
```
зачем пустой инит?

ismail9001 replied:
убери везде где не нужны

makires replied:
Done. Also add custom init through extension.
Commit 9c074b5dc6b04cd95532e4f3cbc01cde90321842 and a81586e7af460a4aebd47802f3cb09a23dddd08b

ismail9001 commented on WeatherProjectTestOrion/Main/Model/Weather.swift:
```diff
@@ -0,0 +1,66 @@
+    }
+}
+
+struct HourlyCurrentWeather {
```
в отдельный файл

makires replied:
Done. 841a43f80828d2c527d42ecf4e4b1826dfabf6e2

ismail9001 commented on WeatherProjectTestOrion/Main/Model/Weather.swift:
```diff
@@ -0,0 +1,66 @@
+        humidity = String(response.current.humidity)
+        feelsLikeTemperature = response.current.feelsLikeTemperature.temperatureConverter
+        text = response.current.condition.text
+        icon = "https:" + response.current.condition.iconURL
```
оно нам нужно?

makires replied:
Рассуждаю по этой же логике.
https://github.com/makires/WeatherProjectTestOrion/pull/2#discussion_r775791770
Модель Weather создается с дефолт значениями в виде пустых срок, эта модель @Published. Как только срабатывает запрос на сервер и приходит ответ, то значения от сервера конвертируются в стринги для модели, соответсвенно для указанных свойств значения устанавливаются через инициализатор и сразу конвертируются. 
Еще момент: temperatureConverter это computed property, в нем сразу добавляется необходимые символы градуса º и + для температуры.

iconURL - не смогла пока продумать как отображает "чистое" небо, если ответ приходит в ночное время.
это тоже к этому комментарию https://github.com/makires/WeatherProjectTestOrion/pull/2#discussion_r775795037

makires replied:
Done. icon for clear day for all variants with icon "sun.max"

ismail9001 commented on WeatherProjectTestOrion/Main/Model/Weather.swift:
```diff
@@ -0,0 +1,66 @@
+        let currentDate = dateFormatter.date(from: response.location.currentLocalTime) ?? Date()
+        var arrayHoursAll: [Hour] = []
+        var arrayHoursPublic: [Hour] = []
+        arrayHoursAll = response.forecast.forecastday[0].hour
```
а если нет нулевого элемента?

makires replied:
Done. Used optional binding to find out whether on optional contains value. Add property .first? and deleted index [0].

ismail9001 commented on WeatherProjectTestOrion/Main/Model/Weather.swift:
```diff
@@ -0,0 +1,66 @@
+        for hour in arrayHoursAll {
+            if Date(timeIntervalSince1970: TimeInterval(hour.timeEpoch)) < currentDate {
+                // nothing do, next iteration
+            } else if Date(timeIntervalSince1970: TimeInterval(hour.timeEpoch)) >= currentDate {
```
если выше ничего не делаем оставь просто этот if

makires replied:
Done.

ismail9001 commented on WeatherProjectTestOrion/Main/View/MainView.swift:
```diff
@@ -0,0 +1,29 @@
+                HeaderWeatherView( weatherVM: weatherVM)
+                DailyWeatherListView(weatherVM: weatherVM)
+            }
+//            .border(.green, width: 2)
```
убирай весь закомментированный код

makires replied:
Done.

ismail9001 commented on WeatherProjectTestOrion/Main/View/MainView.swift:
```diff
@@ -0,0 +1,29 @@
+    @Environment(\.locale.identifier) var locale
+    @ObservedObject var weatherVM = WeatherViewModel(weatherService: WeatherService() )
+    var body: some View {
+            VStack(spacing: 0) {
```
.zero

makires replied:
Перенесен в файл с константами.

ismail9001 commented on WeatherProjectTestOrion/Main/View/MainView.swift:
```diff
@@ -0,0 +1,29 @@
+            }
+//            .border(.green, width: 2)
+            .onAppear {
+                weatherVM.getAllWeather(for: weatherVM.cityTitleStatic, locale: locale)
```
locale можем вынести во viewModel?

makires replied:
На данный момент при размещении locale во ViewModel идентификатор локализации не считывается.

ismail9001 commented on WeatherProjectTestOrion/Main/View/MainView.swift:
```diff
@@ -0,0 +1,29 @@
+    }
+}
+
+struct ContentView_Previews: PreviewProvider {
```
превью удаляй все они нужны только на этапе верстки

makires replied:
Done. All PreviewProvider deleted.
7282760e1c6373b193170b3e518b6790f82f66b4

ismail9001 commented on WeatherProjectTestOrion/Main/View/SubView/CurrentWeatherView.swift:
```diff
@@ -0,0 +1,63 @@
+import SwiftUI
+
+let celsius: LocalizedStringKey = "°"
+let feelsLike: LocalizedStringKey = ", Feels like "
```
выноси все в единый локализационный файл

makires replied:
Done.

ismail9001 commented on WeatherProjectTestOrion/Main/View/SubView/CurrentWeatherView.swift:
```diff
@@ -0,0 +1,63 @@
+                HStack {
+                    Text(weather.temperatureCurrent)
+                }
+                .font(.system(size: 80, weight: .thin))
```
стили сделай через модификаторы

makires replied:
Done.

ismail9001 commented on WeatherProjectTestOrion/Main/View/SubView/CurrentWeatherView.swift:
```diff
@@ -0,0 +1,63 @@
+                // очень длинным и будет коряво отображаться
+                //                  ⬇︎
+                HStack(alignment: .top, spacing: 0) {
+                    VStack {
```
зачем эти VSTack?

makires replied:
unnecessary, deleted.

ismail9001 commented on WeatherProjectTestOrion/Main/View/SubView/CurrentWeatherView.swift:
```diff
@@ -0,0 +1,63 @@
+                .font(.system(size: 12, weight: .regular))
+            }
+            Spacer()
+            AsyncImage(url: URL(string: weather.icon)) { image in
```
мы вроде отказались от загрузки изображений погоды?

makires replied:
Не смогла пока продумать как отображать "чистое" небо, если ответ приходит в ночное время, то по картинке там луна, если день - то солнце. Описание погоды в этом случае имеет одинаковый цифровой код, а как встроить в модель проверку на ночь-день не успела продумать . 
Заметку оставила.

ismail9001 commented on WeatherProjectTestOrion/Main/View/SubView/CurrentWeatherView.swift:
```diff
@@ -0,0 +1,63 @@
+            } placeholder: {
+                ProgressView()
+            }
+            .frame(width: 60, height: 60)
```
подумай как сделать паддингами

makires replied:
Deleted.

ismail9001 commented on WeatherProjectTestOrion/Main/View/SubView/DailyWeatherListView.swift:
```diff
@@ -0,0 +1,86 @@
+        GeometryReader { _ in
+            ScrollView {
+                Color.clear
+                    .opacity(0)
```
не совсем понятна конструкция с пустым цветом и нулевой прозрачностью

makires replied:
Этот элемент необходим для отслеживания координат верхнего левого угла во время скрола. Без него не к чему прикрепить бэкграунд, из которого передается значение координаты, (через структуру PreferenceKey), на основе этого события я скрываю часовой прогноз. Подумать над другим вариантом реализации?
На счет opacity(0), для PR удалю, прозрачность осталась для дальнейшего эксперимента с реализацией исчезающего часового прогноза, так как еще нет итогового варианта.
Commit 7d93b28.

makires replied:
Переделала на другой вариант.

ismail9001 commented on WeatherProjectTestOrion/Main/View/SubView/DailyWeatherListView.swift:
```diff
@@ -0,0 +1,86 @@
+    }
+}
+
+struct DailyWeatherRow: View {
```
в отдельную сабвью

makires replied:
Done. Separate DailyWeatherRow.
Commit 7282760.

ismail9001 commented on WeatherProjectTestOrion/Main/View/SubView/DailyWeatherListView.swift:
```diff
@@ -0,0 +1,86 @@
+        HStack {
+            VStack(alignment: .leading) {
+                Text(dailyForecast.dateEpoch.formattedDay)
+                    .foregroundColor(Color(#colorLiteral(red: 0.24, green: 0.24, blue: 0.26, alpha: 0.6)))
```
используй цвета через ColorSet в ассетах

makires replied:
Done.

ismail9001 commented on WeatherProjectTestOrion/Main/View/SubView/DailyWeatherListView.swift:
```diff
@@ -0,0 +1,86 @@
+                    .foregroundColor(Color(#colorLiteral(red: 0, green: 0, blue: 0, alpha: 1)))
+            }
+            Spacer()
+            let gridItems = [GridItem(spacing: 10), GridItem( spacing: 4), GridItem(alignment: .trailing)]
```
выноси из боди

makires replied:
Done, property gridItems moved out from body.
Commit 7d93b28.

ismail9001 commented on en.lproj/Localizable.strings:
```diff
@@ -0,0 +1,24 @@
+", Feels like " = ", Feels like ";
+" km/h " = " km/h ";
+" hPa" = " hPa";
+"Nizhny Novgorod" = "Nizhny Novgorod";
```
пиши ключи с уникальными Uppercased

пример "FEELS_LIKE"

makires replied:
Done.

makires committed:
creating .gitignore file

makires committed:
.gitignore is now working

makires committed:
learning images deleted. Manually change name image for cloudBackground.

makires committed:
change names for assets images.

makires committed:
Dummy model deleted.

makires committed:
.opacity for left top Color.clear deleted, it was for tests.
Grids moved from body out.
Commented code deleted

makires committed:
Created subview for DailyWeatherRow. Deleted all PreviewProvider.

makires committed:
Deleted Environment property, not need.
Add extension for struct with custom init

lukvan committed:
renamed property for Weather with weather condition.

lukvan committed:
Deleted default init for model and added extension for these model

lukvan committed:
Separate file for HourlyCurrentWeather
* it is model for weather with use in views

lukvan committed:
enum for windDirection for API model.

lukvan committed:
new .gitignore now is working

makires committed:
new .gitignore file

makires committed:
clear gitignire

makires committed:
project is working. Add new .gitignore

makires committed:
Regenerated project branch developer

makires committed:
change user for gitHub and email, test

makires committed:
modifiers fonts for views, add color set into assets.

makires committed:
Change storage for list of cities through @AppStorage. Now it is a simple Array + Extensions.
Add temporary CitiRowViewModel without parallel request to API.
Add temporary button into ListCities for Refresh cities, Add cities, and test through @StateObject.

makires committed:
Deleted old Localization, add through enum. For dynamic cities add LocalizedStringKey.

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/CityRowView.swift:
```diff
@@ -0,0 +1,134 @@
+
+import SwiftUI
+var minTemperature = "+16º"
+var maxTemperature = "15º"
```
зачем они здесь?

makires replied:
Deleted. It was temporary.

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/CityRowView.swift:
```diff
@@ -0,0 +1,134 @@
+    init(weatherService: WeatherRepositoryProtocol) {
+        self.weatherService = weatherService
+    }
+    @Published var weatherCurrent = Weather()
```
переменные обычно инициализируются до основного инициализатора

makires replied:
Deleted.

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/CityRowView.swift:
```diff
@@ -0,0 +1,134 @@
+    }
+}
+struct CityRowView: View {
+//    @State var cityName: String = ""
```
убираем все комментарии

makires replied:
Done.

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/CityRowView.swift:
```diff
@@ -0,0 +1,134 @@
+        VStack {
+            if editList {
+                HStack {
+                    VStack(spacing: 2) {
```
сделай через Magic number

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/CityRowView.swift:
```diff
@@ -0,0 +1,134 @@
+                            .foregroundColor(.red)
+                            .rotationEffect(Angle(degrees: 45))
+                            .frame(width: 44, height: 44)
+                            .padding(EdgeInsets(top: 6, leading: 0, bottom: 6, trailing: 8))
```
попробуй реализовать иначе, много магических чисел

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/CityRowView.swift:
```diff
@@ -0,0 +1,134 @@
+        }
+        .onAppear(perform: {
+            #warning("viewModel должна быть одна, а у тебя отдельная для ячейки")
+            print("появилась ячейка \(cityName)")
```
убираем

makires replied:
Done.

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/CityRowView.swift:
```diff
@@ -0,0 +1,134 @@
+            
+        })
+        .overlay(RoundedRectangle(cornerRadius: 4).strokeBorder(Color("borderCityRow"))
+                    )
```
должно быть на 63 строке

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/CityRowView.swift:
```diff
@@ -0,0 +1,134 @@
+        .overlay(RoundedRectangle(cornerRadius: 4).strokeBorder(Color("borderCityRow"))
+                    )
+        .padding(.horizontal, 16)
+        .padding(.vertical, 4)
```
магические числа

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/CityRowView.swift:
```diff
@@ -0,0 +1,134 @@
+                Image("mappin")
+                    .resizable()
+                    .scaledToFit()
+                    .frame(width: 10.5, height: 15)
```
попробуй через высоту в 16 и aspectRatio

makires replied:
Done. Worked.

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/CityRowView.swift:
```diff
@@ -0,0 +1,134 @@
+            }
+            HStack {
+//                Text(weather.region)
+                Text("Субъект рф")
```
localized

makires replied:
Весь Location приходит от сервера на английском. 
Q: Что делать , если API не поддерживает русский по всем субъектам РФ и другим странам?

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/CityRowView.swift:
```diff
@@ -0,0 +1,134 @@
+//                Text(weather.country)
+                Text("Страна")
+            }
+            .foregroundColor(Color("subText"))
```
сделай цвета через расширение Color

makires replied:
Colors - done.

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/CityRowView.swift:
```diff
@@ -0,0 +1,134 @@
+        }
+        .padding(EdgeInsets(top: 8, leading: 8, bottom: 8, trailing: 0))
+    }
+    var iconAndCurrentTemperature: some View {
```
непонятно название нужно переделать

makires replied:
Renamed.

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/CityRowView.swift:
```diff
@@ -0,0 +1,134 @@
+    }
+    var iconAndCurrentTemperature: some View {
+        HStack(spacing: 16) {
+            Image(systemName: "cloud")
```
картинки тоже через енум расширения сделай

makires replied:
Done.

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/CityRowView.swift:
```diff
@@ -0,0 +1,134 @@
+                Text("|")
+                Text("Southeast")
+                Text("|")
+                Text("10") + Text(Localization.kmH.localized)
```
можно сделать одной конкатинацией _ плюс много статичных данных

makires replied:
Done.

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/CityRowView.swift:
```diff
@@ -0,0 +1,134 @@
+            }
+            Spacer()
+            HStack {
+                Text(minTemperature) + Text("/") + Text(maxTemperature)
```
Text(minTemperature + "/" + maxTemperature)

makires replied:
Конкатенация строк, done.

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/ListLocationsView.swift:
```diff
@@ -0,0 +1,74 @@
+    @Environment(\.locale.identifier) var locale
+    @State var editList = false
+    var body: some View {
+        ScrollView {
```
соблюдай выравнивание

makires replied:
Done.

ismail9001 commented on WeatherProjectTestOrion/CitiesList/View/ListLocationsView.swift:
```diff
@@ -0,0 +1,74 @@
+        }
+    }
+    
+    
```
пустые строки swiftlint должен был об этом сказать

makires replied:
Починила SwiftLint.

ismail9001 commented on WeatherProjectTestOrion/Main/View/SubView/DailyWeatherRow.swift:
```diff
@@ -0,0 +1,38 @@
+                    Image(systemName: dailyForecast.day.condition.iconName)
+                        .foregroundStyle(.blue, .yellow, .blue)
+                    Text(dailyForecast.day.maxTemperatureCelcius.temperatureConverter)
+                        .tracking(0.35)
```
это можно вынести в стиль текста

ismail9001 commented on WeatherProjectTestOrion/Main/View/TestPluginView/TestSwiftUIPluginForFigma/CurrentWeatherViewTP.swift:
```diff
@@ -0,0 +1,40 @@
+            .lineSpacing(16)
+            .offset(x: -106.50, y: 40)
+
+            ZStack {
```
зачем он нужен?

makires replied:
Deleted.

ismail9001 commented on WeatherProjectTestOrion/Main/View/TestPluginView/TestSwiftUIPluginForFigma/DetailsForCurrentWeatherViewTP.swift:
```diff
@@ -0,0 +1,175 @@
+            Text("83%")
+            .font(.caption)
+            .lineSpacing(16)
+            .offset(x: -194, y: 28)
```
если ты используешь столько offset явно чтото не так с версткой

makires replied:
Линий тестовый файл, удален.

ismail9001 commented on WeatherProjectTestOrion/Network/APICurrentWeatherModel.swift:
```diff
@@ -0,0 +1,219 @@
+    }
+}
+
+extension Int {
```
почему расширения находятся в файле с моделями

makires replied:
Перенесла в общий файл с Modifiers+Extension.
Q: файл перегружен. Как организовывать?

makires replied:
Done.

makires committed:
Add Swift Concurrency into project, changed WeatherRepositoryProtocol.

makires committed:
Enum for icon image.  Tasks: change view (details).

makires committed:
Icon Image for current weather, daily weather and cities from enum. Changed model Weather appropriately.
Deleted unnecessary VStask from CurrentWeather View.
Font for Icon CurrentWeather changed.

makires committed:
changed implementation for scroll daily weather forecast through .coordinateSpace and @Published bool value for isScrolled

makires committed:
Done main View, Cities List.

makires committed:
Merge remote-tracking branch 'refs/remotes/origin/developer'

Conflicts:
	WeatherProjectTestOrion/Main/View/SubView/DailyWeatherListView.swift
	WeatherProjectTestOrion/Main/ViewModel/WeatherViewModel.swift

makires committed:
Оставшиеся детали после pull.

makires committed:
Оставшиеся детали перед merge. Отдельные файлы для вспомогательных элементов.
